pkg/chart/chart.go

```diff
@@ -217,8 +228,41 @@ func (c *Chart) ParseDetails(data []byte) (*Details, error) {
 	return details, nil
 }
 
+// InitNetClient returns an HTTP client loading a custom CA if provided (as a secret)
+func (c *Chart) InitNetClient(customCA *CustomCA) (*http.Client, error) {
+	// Get the SystemCertPool, continue with an empty pool on error
+	caCertPool, _ := x509.SystemCertPool()
+	if caCertPool == nil {
+		caCertPool = x509.NewCertPool()
+	}
+
+	// If additionalCA is set, load it
+	if customCA != nil {
+		namespace := os.Getenv("POD_NAMESPACE")
+		caCertSecret, err := c.kubeClient.CoreV1().Secrets(namespace).Get(customCA.SecretKeyRef.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Fatalf("Unable to read the given CA cert: %v", err)
+		}
+
+		// Append our cert to the system pool
+		if ok := caCertPool.AppendCertsFromPEM(caCertSecret.Data[customCA.SecretKeyRef.Key]); !ok {
+			return nil, fmt.Errorf("Failed to append %s to RootCAs", customCA.SecretKeyRef.Name)
+		}
+	}
+
+	// Return Transport for testing purposes
+	return &http.Client{
+		Timeout: time.Second * defaultTimeoutSeconds,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}, nil
+}
+
 // GetChart retrieves and loads a Chart from a registry
-func (c *Chart) GetChart(details *Details) (*chart.Chart, error) {
+func (c *Chart) GetChart(details *Details, netClient HTTPClient) (*chart.Chart, error) {
```

Comment:
Actually GetChart will read the secret twice in the case of both header and CA are set.

Would it be possible to preload this or cache it?